test/realtime_box_tests.cpp

```diff
@@ -29,36 +30,204 @@
 #include <gmock/gmock.h>
 #include <realtime_tools/realtime_box.h>
 
+struct DefaultConstructable
+{
+  int a = 10;
+  std::string str = "hallo";
+};
+
+struct NonDefaultConstructable
+{
+  NonDefaultConstructable(int a_, const std::string & str_) : a(a_), str(str_) {}
+  int a;
+  std::string str;
+};
+
+struct FromInitializerList
+{
+  FromInitializerList(std::initializer_list<int> list)
+  {
+    std::copy(list.begin(), list.end(), data.begin());
+  }
+  std::array<int, 3> data;
+};
+
 using realtime_tools::RealtimeBox;
 
-class DefaultConstructable
+TEST(RealtimeBox, empty_construct)
+{
+  RealtimeBox<DefaultConstructable> box;
+
+  auto value = box.get();
+  EXPECT_EQ(value.a, 10);
+  EXPECT_EQ(value.str, "hallo");
+}
+
+TEST(RealtimeBox, default_construct)
+{
+  DefaultConstructable data;
+  data.a = 100;
+
+  RealtimeBox<DefaultConstructable> box(data);
+
+  auto value = box.get();
+  EXPECT_EQ(value.a, 100);
+  EXPECT_EQ(value.str, "hallo");
+}
+
+TEST(RealtimeBox, non_default_constructable)
+{
+  RealtimeBox<NonDefaultConstructable> box(NonDefaultConstructable(-10, "hello"));
+
+  auto value = box.get();
+  EXPECT_EQ(value.a, -10);
+  EXPECT_EQ(value.str, "hello");
+}
+TEST(RealtimeBox, standard_get)
+{
+  RealtimeBox<DefaultConstructable> box(DefaultConstructable{.a = 1000});
+
+  DefaultConstructable data;
+  box.get(data);
+  EXPECT_EQ(data.a, 1000);
+  data.a = 10000;
+
+  box.set(data);
+
+  auto value = box.get();
+  EXPECT_EQ(value.a, 10000);
+}
+
+TEST(RealtimeBox, initializer_list)
+{
+  RealtimeBox<FromInitializerList> box({1, 2, 3});
+
+  auto value = box.get();
+  EXPECT_EQ(value.data[0], 1);
+  EXPECT_EQ(value.data[1], 2);
+  EXPECT_EQ(value.data[2], 3);
+}
+
+TEST(RealtimeBox, assignment_operator)
+{
+  DefaultConstructable data;
+  data.a = 1000;
+  RealtimeBox<DefaultConstructable> box;
+  // Assignment operator is always non RT!
+  box = data;
+
+  auto value = box.get();
+  EXPECT_EQ(value.a, 1000);
+}
+TEST(RealtimeBox, typecast_operator)
+{
+  RealtimeBox<DefaultConstructable> box(DefaultConstructable{.a = 100, .str = ""});
+
+  // Use non RT access
+  DefaultConstructable data = box;
+
+  EXPECT_EQ(data.a, 100);
+
+  // Use RT access -> returns std::nullopt if the mutex could not be locked
+  std::optional<DefaultConstructable> rt_data_access = box;
+
+  if (rt_data_access) {
+    EXPECT_EQ(rt_data_access->a, 100);
+  }
+}
+
+TEST(RealtimeBox, pointer_type)
+{
+  int a = 100;
+  int * ptr = &a;
+
+  RealtimeBox<int *> box(ptr);
+  // This does not and should not compile!
+  // auto value = box.get();
```

Comment:
```suggestion
```

Reply:
At the moment this would compile btw. (Because we added the pointer specialisation for compatibility reasons)

Reply:
Sure, then let's remove the comment.